### PR TITLE
Mounted AC rework

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
Mounted autocannon rework. It now spawns as a moveable gun like the TAT, it is not roundstart if you're wondering..
It is now dedicated wall destroyer, dealing bonus true damage to walls, mostly effective against softer wall types like resin walls.

Overall, it fires faster, has a FULLER AUTO MODE, and the gun is balanced for heavy rifle firerate alongside the fuller auto.

Further in detail belowl
<details>

Left is new|Right is old.

Gun
-

Has burst fire, 12 round. .25 firerate, has some level of scatter.
Fire delay 0.4|0.5
Scope upgraded to TAT level.
Armor, coverage upgraded to TAT level.
Still has IFF.

Ammo
-

In general now has a minimum accurate range of 5 tiles, and does true damage to walls.

Normal
Damage profile of 30/50/1 from 40/100/8. Does 15 true damage to walls.

Flak
Damage profile of 40/20/5 from 40/20/12. Does 5 true damage to walls.

PASS 1
Autoburst reworked into burst fire, should make it shred things less.

</details>

Sprites
image

by crow, so credit to him.
## Why It's Good For The Game
Makes the autocannon unique and have a niche at the same time in both wall destruction and very long range fire support.
## Changelog
:cl:
balance: Autocannon reworked, it now has a new sprite by Crow, it now does bonus damage to walls, it does much less sunder and damage, now fires much faster and with an autoburst.
/:cl:

### CODE REVIEW

**Comment by Pariah919** [comment]
#12595  is better

**Comment by maikilangiolo** [comment]
> Fire delay 0.5|0.25

Old fire delay was 0.5, not 0.25

Anyway, this needs a TM because it's promising

**Review by Kuro020 — APPROVED** [review]

**Comment by iforgotmeword** [comment]
I call for the gun being nerfed or limited by making it AT OR AC 

marines are kind of getting a lot of wall breaking tools already

**Comment by Lumipharon** [comment]
This is a significant dps buff even before calculating fullerauto. @Pariah919

**Comment by Pariah919** [comment]
> This is a significant dps buff even before calculating fullerauto. @Pariah919

Yeah I suppose so, you think upping the fire delay to like .35 or .4 would be good with the fuller auto included?

**Comment by srxtreme** [comment]
This makes the autocannon (an expensive deployable like TAT), worse at damaging a Queen (high armor target, 65 bullet at ancient) than an AR11 (a rifle).
If the goal of the PR was to make autocannon a joke then it was a glorious success.

**Comment by maikilangiolo** [comment]
After the changes requested by Lumi, this has turned in a major nerf PR. It does less damage than before (yes, even per second: 80 (before) and 75 (now)), it doesn't even do true damage vs high armor castes like before, it does less sunder (2.5 vs 16), the burst is quite honestly not worth using. The spread makes you waste half and more (depending on !fun! RNG) of your (very limited) shots at just 5 tiles of range. The fall off has risen to 5% out of 30 damage, laughable now that it doesn't even have the initial ROF to back it up.

Either maints see reason without requesting inane number changes at a whim, or please, MJP, close this PR. If you made this gun 500 points cheaper and give it larger mags, maybe it would be serve a purpose from time to time. As it stands, it's out done in damage and sunder by almost every gun with no purpose (the wall damage is too little and not worth for such a small magazine).

Fyi, the HSG still does 160 DPS 40 AP 20 SPS, vs this neutered and honestly quite sad 75 DPS 50 AP 2.5 SPS. And it's free roundstart, cheaper, with bigger mags and can be picked up. It also breaks resin walls better.

File: code/modules/projectiles/guns/mounted.dm [hunk 1]
```diff
@@ -165,27 +165,33 @@
 
 	scatter = 10
 	deployed_scatter_change = -10
-	fire_delay = 0.5 SECONDS
+	fire_delay = 0.4 SECONDS
+	burst_amount = 12
```
Comment (Lumipharon) [inline code/modules/projectiles/guns/mounted.dm:169]: probs need a lower burst scatter mult, or just lower scatter in general. The burst scatter scales with burst amount, so its pretty high.

File: code/modules/projectiles/guns/mounted.dm [hunk 2]
```diff
@@ -165,27 +165,33 @@
...
+	burst_delay = 0.25 SECONDS
```
Comment (Lumipharon) [inline code/modules/projectiles/guns/mounted.dm:170]: could probs make this lower. The long delay between bursts means you get punished pretty bad just shooting wildly, and its a little slow.

File: code/modules/projectiles/ammo_datums.dm [hunk 3]
```diff
@@ -1284,16 +1290,20 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 /datum/ammo/bullet/auto_cannon/flak
 	name = "autocannon smart-detonating bullet"
 	hud_state = "sniper_flak"
-	flags_ammo_behavior = AMMO_BALLISTIC|AMMO_SUNDERING|AMMO_PASS_THROUGH_TURF
+	flags_ammo_behavior = AMMO_BALLISTIC|AMMO_SUNDERING|AMMO_PASS_THROUGH_TURF|AMMO_EXPLOSIVE
```
Comment (Lumipharon) [inline code/modules/projectiles/ammo_datums.dm:1293]: Why does flak pass through walls lol?

**Review by Lumipharon — CHANGES_REQUESTED** [review]

**Pull request closed without merging**